backend/src/main/java/com/woowacourse/friendogly/club/domain/Club.java

```diff
@@ -126,6 +137,78 @@ public static Club create(
                 .createdAt(LocalDateTime.now())
                 .imageUrl(imageUrl)
                 .build();
+        club.addClubMember(club.owner);
+        return club;
+    }
+
+    public void addClubMember(Member newMember) {
+        validateAlreadyExists(newMember);
+        validateMemberCapacity();
+        ClubMember clubMember = ClubMember.create(this, newMember);
+        clubMembers.add(clubMember);
+        clubMember.updateClub(this);
+    }
+
+
+    private void validateAlreadyExists(Member newMember) {
+        if (clubMembers.stream()
+                .anyMatch(clubMember -> Objects.equals(clubMember.getMember().getId(), newMember.getId()))
+        ) {
+            throw new FriendoglyException("이미 참여 중인 모임입니다.");
+        }
+    }
+
+    private void validateMemberCapacity() {
+        if (clubMembers.size() >= memberCapacity.getValue()) {
+            throw new FriendoglyException("최대 인원을 초과하여 모임에 참여할 수 없습니다.");
+        }
+    }
+
+    public void addClubPet(List<Pet> pets) {
+        pets.stream()
+                .peek(this::validateParticipatePet)
+                .map(pet -> new ClubPet(this, pet))
+                .peek(clubPet -> clubPet.updateClub(this))
+                .forEach(clubPets::add);
```

Comment:
```suggestion
List<ClubPet> clubPets = pets.stream()
                .peek(this::validateParticipatePet)
                .map(pet -> new ClubPet(this, pet))
                .peek(clubPet -> clubPet.updateClub(this))
                .toList();
        this.clubPets.addAll(clubPets);
```

ArrayList는 요소 추가 시 가변적으로 배열 길이를 조절합니다. 
만약 많은 수의 ClubPet을 처리해야 한다면, 최대 배열 길이를 넘어선 add연산이 수행될 때마다 새로운 배열을 생성해 값을 복사해 넣습니다. 
따라서 foreach에서 add 연산을 수행하는 것 보단, 중간 연산 결과를 List로 모으고 한 번에 리스트에 추가하는 편이 좋아보입니다!
물론 ArrayList 내부의 배열 크기를 넘어설 만큼의 ClubPet이 입력될 일은 없을테니 참고만 해주세요.